src/open_inwoner/accounts/tests/test_profile_views.py

```diff
@@ -346,6 +363,63 @@ def test_expected_form_is_rendered(self):
 
         self.assertEqual(type(form), BrpUserForm)
 
+    def test_image_is_saved_when_begeleider_and_default_login(self):
+        self.user.contact_type = ContactTypeChoices.begeleider
+        self.user.save()
+
+        img_bytes = self.create_test_image_bytes()
+        form_response = self.upload_test_image_to_profile_edit_page(img_bytes)
+
+        self.assertRedirects(form_response, reverse("accounts:my_profile"))
+        with self.assertRaises(ValueError):
+            self.user.image.file
+
+        self.user.refresh_from_db()
+
+        self.assertIsNotNone(self.user.image.file)
+
+    def test_image_is_saved_when_begeleider_and_digid_login(self):
+        self.user.contact_type = ContactTypeChoices.begeleider
+        self.user.login_type = LoginTypeChoices.digid
+        self.user.save()
+
+        img_bytes = self.create_test_image_bytes()
+        form_response = self.upload_test_image_to_profile_edit_page(img_bytes)
+
+        self.assertRedirects(form_response, reverse("accounts:my_profile"))
+        with self.assertRaises(ValueError):
+            self.user.image.file
+
+        self.user.refresh_from_db()
+
+        self.assertIsNotNone(self.user.image.file)
+
+    def test_image_is_not_saved_when_no_begeleider_and_default_login(self):
+        img_bytes = self.create_test_image_bytes()
+        with self.assertRaises(AssertionError) as e:
+            self.upload_test_image_to_profile_edit_page(img_bytes)
```

Comment:
Why would this raise an AssertionError, and why is this assertion new in the test? I was looking at the diff of the commit but I don't see anything like that in the code this replaces. I also don't see anything in the new helpers that could cause this.

Also (related), in general: if you use `assertRaises` it can be very readable to use `assertRaisesMessage` or `assertRaisesRegex` so it is more clear what kind of error happens. Sometimes its obvious (like the ValueError's) but like here it is hard to determine.

Reply:
I added the `AssertionError` because I changed the form init method (now if the user is not a begeleider I delete the field from the form, as you suggested in your previous comment, so it cannot find the field in the form).